timApp/answer/routes.py

```diff
@@ -2337,12 +2337,36 @@ def get_model_answer(task_id: str) -> Response:
     if not model_answer_info or not model_answer_info.answer:
         raise RouteException(f"No model answer for task {task_id}")
     if not has_teacher_access(d):
+        answer_count: Union[int | None] = None
+        if model_answer_info.disabled:
+            raise RouteException("This model answer has been disabled")
+        if model_answer_info.endDate:
+            if model_answer_info.endDate < get_current_time():
+                raise RouteException("This model answer is no longer accessible")
         if model_answer_info.revealDate:
             if model_answer_info.revealDate > get_current_time():
                 raise RouteException("The model answer cannot be viewed yet")
+        if model_answer_info.groups and len(model_answer_info.groups) > 0:
```

Comment:
```suggestion
        if model_answer_info.groups:
```

koska `bool(lista) <=> lista is not None and len(lista) > 0`